zendriver/core/connection.py

```diff
@@ -230,7 +230,7 @@ def target(self, target: cdp.target.TargetInfo):
     def closed(self):
         if not self.websocket:
             return True
-        return self.websocket.closed
+        return (not self.websocket)
```

Comment:
I see you added
```py
websocket = None
```
below, so this will work when _we_ close the connection, but aren't there also other ways for the websocket to get closed? what if the browser closes the connection?

to handle all cases, we should probably do like

```
self.websocket is None or self.websocket.closed
```

(this applies to the other places you changed this as well)

Reply:
closed property removed in 14 and logic of websockets authors : 
closed can be changed only on operations (send/recv) - here no any loop in websockets that check sockets in background
closed attribute can be filled only on operations, that already raise ConnectionClosedError
using of closed attribute after this exception is bad design for applications